sklearn/semi_supervised/tests/test_label_propagation.py

```diff
@@ -173,9 +174,9 @@ def test_label_propagation_non_zero_normalizer(LabelPropagationCls):
     X = np.array([[100.0, 100.0], [100.0, 100.0], [0.0, 0.0], [0.0, 0.0]])
     y = np.array([0, 1, -1, -1])
     mdl = LabelPropagationCls(kernel="knn", max_iter=100, n_neighbors=1)
-    with pytest.warns(None) as record:
+    with warnings.catch_warnings():
+        warnings.simplefilter("error", UserWarning)
```

Comment:
Looking over the PR [#15946](https://github.com/scikit-learn/scikit-learn/pull/15946) linked a few lines above, I think this should be a run time warning:

```suggestion
        warnings.simplefilter("error", RuntimeWarning)
```

Reply:
Thank you @thomasjpfan, I have added your suggestion